File: backend/app/schemas/ask.py
from pydantic import BaseModel, Field
from typing import Optional, List
from datetime import datetime
from enum import Enum

class AskCategory(str, Enum):
    ACADEMIC = "ACADEMIC"
    ITEMS = "ITEMS"
    FOOD = "FOOD"
    TRANSPORT = "TRANSPORT"
    LOCATION = "LOCATION"
    EMERGENCY = "EMERGENCY"
    EVENT = "EVENT"
    OTHER = "OTHER"

class AskCreate(BaseModel):
    title: str = Field(..., min_length=3, max_length=100)
    description: str = Field(..., min_length=5, max_length=500)
    category: AskCategory
    location: str = Field(..., min_length=2, max_length=100)
    expires_in_minutes: int = Field(default=20, ge=5, le=1440)  # Max 24 hours

class AskResponse(BaseModel):
    id: str
    requester_id: str
    requester_name: str
    requester_image: str
    title: str
    description: str
    category: str
    location: str
    status: str
    reply_count: int
    max_replies: int
    created_at: str
    expires_at: str

class ReplyCreate(BaseModel):
    message: str = Field(..., min_length=1, max_length=500)

class ReplyResponse(BaseModel):
    id: str
    ask_id: str
    responder_id: str
    responder_name: str
    responder_image: str
    message: str
    created_at: str

class AskResolve(BaseModel):
    reply_id: str

File: frontend/lib/models/ask.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'ask.freezed.dart';
part 'ask.g.dart';

@freezed
class Ask with _$Ask {
  const factory Ask({
    required String id,
    required String requester_id,
    required String requester_name,
    required String requester_image,
    required String title,
    required String description,
    required String category,
    required String location,
    required String status,
    required int reply_count,
    required int max_replies,
    required String created_at,
    required String expires_at,
  }) = _Ask;

  factory Ask.fromJson(Map<String, dynamic> json) => _$AskFromJson(json);
}

@freezed
class Reply with _$Reply {
  const factory Reply({
    required String id,
    required String ask_id,
    required String responder_id,
    required String responder_name,
    required String responder_image,
    required String message,
    required String created_at,
  }) = _Reply;

  factory Reply.fromJson(Map<String, dynamic> json) => _$ReplyFromJson(json);
}
